Ginger/GingerCoreCommon/Repository/RepositoryFolder.cs

```diff
@@ -696,12 +696,15 @@ public override RepositoryFolderBase AddSubFolder(string folderName)
         /// Save the Repository Item to folder and add it to cache
         /// </summary>
         /// <param name="repositoryItem"></param>
-        public override void AddRepositoryItem(RepositoryItemBase repositoryItem)
+        public override void AddRepositoryItem(RepositoryItemBase repositoryItem, bool DoNotSave = false)
         {
-            //save it
-            repositoryItem.ContainingFolder = FolderRelativePath;
-            repositoryItem.ContainingFolderFullPath = FolderFullPath;
-            SolutionRepository.SaveNewRepositoryItem(repositoryItem);
+            if (!DoNotSave)
+            {
+                //save it
+                repositoryItem.ContainingFolder = FolderRelativePath;
```

Comment:
do we have an issue to still include the folder details on the object? i know it not going to be saved so it misleading but maybe we have functionality somewhere that pulls the folder from the object for some need so if not causing any known issue so let's